File: bindings/lua/Polycode/RendererThreadJob.lua
class "RendererThreadJob"


function RendererThreadJob:__getvar(name)
	if name == "jobType" then
		return Polycode.RendererThreadJob_get_jobType(self.__ptr)
	end
end

function RendererThreadJob:__setvar(name,value)
	if name == "jobType" then
		Polycode.RendererThreadJob_set_jobType(self.__ptr, value)
		return true
	end
	return false
end
function RendererThreadJob:__delete()
	if self then Polycode.delete_RendererThreadJob(self.__ptr) end
end

File: bindings/javascript/Polycode/RendererThreadJob.js
function RendererThreadJob() {
	if(arguments[0] != "__skip_ptr__") {
		this.__ptr = Polycode.RendererThreadJob()
	}
	Object.defineProperties(this, {
		'jobType': { enumerable: true, configurable: true, get: RendererThreadJob.prototype.__get_jobType, set: RendererThreadJob.prototype.__set_jobType}
	})
}


RendererThreadJob.prototype.__get_jobType = function() {
	return Polycode.RendererThreadJob__get_jobType(this.__ptr)
}

RendererThreadJob.prototype.__set_jobType = function(val) {
	Polycode.RendererThreadJob__set_jobType(this.__ptr, val)
}

Duktape.fin(RendererThreadJob.prototype, function (x) {
	if (x === RendererThreadJob.prototype) {
		return;
	}
	Polycode.RendererThreadJob__delete(x.__ptr)
})
